### DISCUSSION TITLE
Clover Doesn't See a PCIe NVMe on System with BIOS

### DISCUSSION THREAD

**Discussion by Gonochnik** [CloverHackyColor/CloverBootloader#556] [Q&A]
Hello to all.
System: Gigabyte GA-X58-Extreme Motherboard with Intel X5690 Processor + ATI RADEON HD 5770 Video.
SSD: Samsung NVMe 970 EVO Plus with PCIe X4 Adaptor + SATA SanDisk. In Windows PCIe NVMe shows correctly without any errors.

Main task to Install a new copy of Windows 11 to the new NVMe and after format the Sandisk SSD with old 11 and Install Hackintosh to it.

What I did. 
Last version of Clover (к5149) has been installed through BootDiskUtility (FAT32 LBA, Boot Partition 200).
NvmExpressDxe was copied from Efi\CLOVER\drivers\off to Efi\CLOVER\drivers\BIOS.
First time after PC was rebooted, Clover didn't Start. I changed Boot6 to Boot7 and it sussesfully loaded.
After I go to F3 options I see only USB FlashCard with Clover & old SATA SSD with Windows. No PCIe NVMe.
Same drives I see in Clover UEFI Shell 64 section.
Tryed to Start Windows 11 on the old SSD through the Clover. Boots without problems.

After I red some Titorials I tryed to copy NvmExpressDxe to Efi\CLOVER\drivers\UEFI (strange action because PC hasn't UEFI), made 2 directories drivers32uefi & drivers64uefi in the Efi\CLOVER directory and copied NvmExpressDxe to them... Same story - Clover BootLoader doesn't see PCIe NVMe.

What shall I do to load NVMe Driver to have access to the Drive? May be I need to add some sections to the config.plist?

Any ideas, counterparts?

ps: Клевер цвета хаки тоже изучил.

**Comment by Gonochnik** [comment]
upd: I took a modified boot7 (CloverBootloader-5146-NVMe.zip) & NvmExpressDxe (NvmExpressDxe_Small.zip) from this thread https://winraid.level1techs.com/t/how-to-get-a-apple-oem-ssd-to-boot-in-a-normal-pc/33205 (thanks to Ethaniel), replaced clover originals with them, reboot & NVMe SSD detected. 
After I made a boot USB with Windows 11 (use Universal Media Creation Tool), installed it succesfully and now Windows 11 on GPT NVMe boot & works without any errors.

**Comment by 1smokinjoe** [comment]
you need to know from the clover team what version of NVME they support.  I could use a Samsung 970 EVO NVME 1.3 disk on a HP Z800 workstation but not a WD Blue 1TB NVME 1.4 disk, clover would not see the 1.4 disk.  At lest that is the theory I have, maybe the samsung is just better and easier for clover to detect where the WD blue uses the cheapest junk from china to make a NVME disk....
